### PR TITLE
Consistent accounts terminology (For budget / Budgeted --> On budget)

### PR DESCRIPTION
**Pull request by joel-jeremy**
Accounts that are on budget are either referred to as `For budget` / `Budgeted` in the UI and the codebase. This PR aims to just refer to these accounts as `on-budget` for consistency.

Changes are also reflected in docs https://github.com/actualbudget/docs/pull/504

### CODE REVIEW

File: packages/desktop-client/src/components/mobile/accounts/Accounts.tsx [hunk 1]
```diff
@@ -178,7 +178,7 @@ type AccountListProps = {
   accounts: AccountEntity[];
   updatedAccounts: Array<AccountEntity['id']>;
   getBalanceQuery: (account: AccountEntity) => Binding<'account', 'balance'>;
-  getOnBudgetBalance: () => Binding<'account', 'budgeted-accounts-balance'>;
+  getOnBudgetBalance: () => Binding<'account', 'onbudget-accounts-balance'>;
   getOffBudgetBalance: () => Binding<'account', 'offbudget-accounts-balance'>;
```
[the remark this answers is not in the document]
Comment (joel-jeremy) [reply packages/desktop-client/src/components/mobile/accounts/Accounts.tsx:182]: This suggestion is wrong. They're both lowercase.

**Comment by matt-fidd** [comment]
I can't nail it down to a grammatical rule, but I'm not sure about the hyphenated version. It looks cleaner to me as: "On budget"

What do you think?

**Comment by joel-jeremy** [comment]
Sure we can do the unhyphenated version. As long as it's consistent across the app and docs :)

**Comment by matt-fidd** [comment]
Looks good! Happy to approve when VRT is rerun

**Review by matt-fidd — APPROVED** [review]
LGTM!

**Pull request merged**